src/main/java/org/jitsi/jicofo/recording/jibri/JibriSession.java

```diff
@@ -300,9 +253,34 @@ private void sendStopIQ(XMPPError error)
 
         logger.info("Trying to stop: " + stopRequest.toXML());
 
-        xmpp.sendStanza(stopRequest);
+        // When we send stop, we won't get an OFF presence back (just
+        // a response to this message) so clean up the session
+        // in the processing of the response.
+        try
+        {
+            xmpp.sendIqWithResponseCallback(stopRequest, stanza -> {
```

Comment:
what happens when there is a timeout ? does the sendIqWithResponseCallback method throw a timeout exception ? If yes, then is it handled ?